src/data/simple_csr_source.cc

```diff
@@ -123,15 +123,29 @@ const SparsePage& SimpleCSRSource::Value() const {
 /*!
  * Please be careful that, in official specification, the only three required fields are
  * `shape', `version' and `typestr'.  Any other is optional, including `data'.  But here
- * we have two additional requirements for input data:
+ * we have one additional requirements for input data:
  *
  * - `data' field is required, passing in an empty dataset is not accepted, as most (if
  *   not all) of our algorithms don't have test for empty dataset.  An error is better
  *   than a crash.
  *
- * - `null_count' is required when `mask' is presented.  We can compute `null_count'
- *   ourselves and copy the result back to host for memory allocation.  But it's in the
- *   specification of Apache Arrow hence it should be readily available,
+ * Missing value handling:
```

Comment:
What if we supported mask and missing value being specified simultaneously?

Reply:
@RAMitchell This is in consistent with current constructor from numpy.  We can make changes to it, but that will require a discussion for breaking behaviour.  Right now I just want to be consistent ...